tests/unit/other.tcl

```diff
@@ -258,6 +258,22 @@ start_server {tags {"other"}} {
         lappend res [r get foo]
     } {12 12}
 
+    test {APPEND modifies the encoding from int to raw} {
+        r del foo
+        r set foo 1
+        r append foo 2
+
+        # append command undergoes a operation dbUnshareStringValue which
+        # converts the value to a sds object of encoding raw type.  
+        assert_equal 12 [r get foo]
+        assert_encoding "raw" foo
+
+        r set bar 12
+        assert_encoding "int" bar
```

Comment:
```suggestion
        assert_encoding "int" foo
        r append foo 2

        assert_equal 12 [r get foo]
        assert_encoding "raw" foo

        r set bar 12
        assert_encoding "int" bar
```
I'm afraid the documentation will fall out of date by mention dbUnshareStringValue, besides that it's basically restating the test. Also worth pre-validating the type is int.